File: apps/mobile/ios/MarkWise/MLKitCameraManager.m
#import <React/RCTViewManager.h>

@interface RCT_EXTERN_MODULE(MLKitCameraViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(onBarcodeScan, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onCameraReady, RCTDirectEventBlock)
@end

File: apps/mobile/ios/MarkWise/MLKitCameraManager.swift
import React

/// RCTViewManager that vends MLKitCameraViewImpl as the native component for
/// requireNativeComponent('MLKitCameraView').  The "Manager" suffix is stripped
/// by React Native's bridge discovery to expose the component as "MLKitCameraView".
@objc(MLKitCameraViewManager)
class MLKitCameraViewManager: RCTViewManager {
  override func view() -> UIView! {
    MLKitCameraViewImpl()
  }

  override static func requiresMainQueueSetup() -> Bool { true }
}
